kernel/src/process.rs

```diff
@@ -261,7 +267,7 @@ impl<'a> Process<'a> {
             breakpoint();
         }
         write_volatile(&mut SYSCALL_FIRED, 0);
-        let psp = switch_to_user(self.cur_stack, self.memory.as_ptr());
+        let psp = switch_to_user(self.cur_stack, self.memory.as_ptr(), &mut self.stored_regs);
```

Comment:
If the security risk here is that the process stack pointer is untrusted, aren't uses of it, such as the `pop_syscall_stack`, `push_callback`, `svc_number` and particular register reading functions also a risk? Or would the `svc` already have led to a hardfault before the kernel took over if the PSP was invalid?